datastream-server/src/main/java/com/linkedin/datastream/server/zk/ZkAdapter.java

```diff
@@ -897,6 +899,28 @@ public void acquireTask(DatastreamTaskImpl task, Duration timeout) {
     }
   }
 
+  /**
+   * Check if the task is current locked
+   */
+  public boolean checkIfTaskLocked(String connectorType, String taskName) {
+    String lockPath = KeyBuilder.datastreamTaskLock(_cluster, connectorType, taskName);
+    return (_zkclient.exists(lockPath));
+  }
+
+  /**
+   * Wait for all dependencies to be cleared
+   * @param task Datastream task whose dependencies need to be checked
+   * @param timeout max wait time to wait for a locked task for releasing
+   */
+  public void waitForDependencies(DatastreamTaskImpl task, Duration timeout) {
```

Comment:
Might want to add a comment saying its a blocking call.

Reply:
Np